### PR TITLE
[WIP] Add ConsoleNotification CRD that can appear above, below, or above and below the page with option to dismiss

### PR DESCRIPTION
**Pull request by rhamilto**
Differs from https://github.com/openshift/console/pull/1542 in that there is a dismiss option.

TODOS:
- [ ] Remove `extensions/console-notification.crd.yaml` from PR as it will live in the console operator

localhost_9000_k8s_cluster_console openshift io_v1_ConsoleNotification_example_yaml (1)

### CODE REVIEW

**Comment by mbach04** [comment]
Will the notification background color be configurable as well? The classification banner looks great, but it's probably best to not keep the standard green success banner notification directly under it as green generally denotes Unclassified content. If that color could be made blue like the oauth message, it wouldn't conflict with any network banner color.

**Comment by spadgett** [comment]
Hey @mbach04, the green banner is another custom message with different options. It's not a standard console notification.

Background color is configurable.

File: extensions/console-notification.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+              - href
+              properties:
+                text:
+                  type: string
+                  description: Text of the link
+                href:
+                  type: string
+                  description: Absolute URL for the link
+                  pattern: '^http(s)?://([\w-]+.)+[\w-]+(/[\w- ./?%&=])?$'
+                opensNewWindow:
```
Comment (rhamilto) [inline extensions/console-notification.crd.yaml]: Let's remove this and make these always open in a new window.

File: frontend/public/components/console-notifier.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        style={{
+          backgroundColor: notification.spec.backgroundColor,
+          color: notification.spec.color,
+        }}>
+        <div className="co-global-notification__content">
+          {notification.spec.dismissible && <CloseButton onClick={() => dismiss(notification.metadata.uid)} />}
+          <p className="co-global-notification__text">
+            {notification.spec.text} {_.get(notification.spec, ['link', 'href'])
+              && <a href={notification.spec.link.href}
+                target={notification.spec.link.opensNewWindow ? '_blank' : null}
```
Comment (rhamilto) [inline frontend/public/components/console-notifier.tsx]: Add` rel="noopener nofollow"`

**Comment by rhamilto** [comment]
Closing in favor of #1542

**Pull request closed without merging**